android_10/src/org/ros/android/MasterChooser.java

```diff
@@ -348,6 +372,52 @@ protected boolean isQRCodeReaderInstalled(Intent intent) {
   }
 
   /**
+   * Adds the given URI to the list of recent Master URIs stored in shared preferences.
+   * This implementation does not use
+   * {@link android.content.SharedPreferences.Editor#putStringSet(String, Set)}
+   * since it is not available in API 10.
+   * @param uri Master URI string to store.
+   */
+  private void setRecentMasterURI(String uri) {
+    List<String> recentURIs = getRecentMasterURIs();
+    if(!recentURIs.contains(uri)) {
```

Comment:
Another formatting nit: space between `if` and open parenthesis (and `for`, etc.).